src/v/cloud_storage/cache_service.cc

```diff
@@ -174,23 +186,92 @@ ss::future<> cache::clean_up_cache() {
     }
 }
 
+ss::future<> cache::load_access_time_tracker() {
+    ss::gate::holder guard{_gate};
+    vassert(ss::this_shard_id() == 0, "Method can only be invoked on shard 0");
+    auto source = _cache_dir / access_time_tracker_file_name;
+    auto present = co_await ss::file_exists(source.native());
+    if (!present) {
+        vlog(cst_log.info, "Access time tracker doesn't exist at '{}'", source);
+        co_return;
+    }
+    vlog(
+      cst_log.info, "Trying to hydrate access time tracker from '{}'", source);
+    if (auto cache_item = co_await get(source); cache_item.has_value()) {
+        try {
+            ss::file_input_stream_options options{};
+            options.buffer_size
+              = config::shard_local_cfg().storage_read_buffer_size;
+            options.read_ahead
+              = config::shard_local_cfg().storage_read_readahead_count;
+            options.io_priority_class
+              = priority_manager::local().shadow_indexing_priority();
+            auto inp_stream = ss::make_file_input_stream(
+              cache_item->body, options);
+            iobuf state;
+            auto out_stream = make_iobuf_ref_output_stream(state);
+            co_await ss::copy(inp_stream, out_stream).finally([&inp_stream] {
+                return inp_stream.close();
+            });
+            _access_time_tracker.from_iobuf(std::move(state));
+        } catch (...) {
+            vlog(
+              cst_log.warn,
+              "Failed to materialize access time tracker '{}'. Error: {}",
+              source,
+              std::current_exception());
+        }
+        co_await cache_item->body.close();
+    } else {
+        vlog(
+          cst_log.info, "Access time tracker is not available at '{}'", source);
+    }
+}
+
+ss::future<> cache::save_access_time_tracker() {
+    ss::gate::holder guard{_gate};
+    vassert(ss::this_shard_id() == 0, "Method can only be invoked on shard 0");
+    auto source = _cache_dir / access_time_tracker_file_name;
+    auto index_stream = make_iobuf_input_stream(
+      _access_time_tracker.to_iobuf());
+    co_await put(source.native(), index_stream);
+}
+
+ss::future<> cache::maybe_save_access_time_tracker() {
+    vassert(ss::this_shard_id() == 0, "Method can only be invoked on shard 0");
+    if (_access_time_tracker.is_dirty() && !_gate.is_closed()) {
+        co_await save_access_time_tracker();
```

Comment:
Are we saving/serializing the LRU state to work around the lack of filesystem access time tracking (which can be expensive; I'd expect us to leave this disabled?)

Reply:
yes, we can't rely on FS to track access times
what do you mean by leaving this disabled?

Reply:
Just remembering that filesystems can track access time for you, but that introduces writes/updates in the read path, which hurts FS performance. Usually it is avoided/disabled (i.e. `noatime` mount option) when performance matters.